src/modules/xworkspaces.cpp

```diff
@@ -86,21 +86,26 @@ namespace modules {
     m_current_desktop_name = m_desktop_names[m_current_desktop];
 
     rebuild_desktops();
-    rebuild_desktop_states();
 
     // Get _NET_CLIENT_LIST
     rebuild_clientlist();
+
+    rebuild_desktop_states();
   }
 
   /**
    * Handler for XCB_PROPERTY_NOTIFY events
    */
   void xworkspaces_module::handle(const evt::property_notify& evt) {
-    if (evt->atom == m_ewmh->_NET_CLIENT_LIST) {
+    std::lock_guard<std::mutex> lock(m_event_mutex);
```

Comment:
```suggestion
    std::lock_guard<std::mutex> lock(m_buildlock);
```
The `m_buildlock` mutex is used to protect the `build` phase and is locked before the call of `build`, so this mutex can be removed.
https://github.com/polybar/polybar/blob/cb75857ff66543e9aedac3f87df130ba03132471/include/modules/meta/base.inl#L124-L176
The lock is at line 125 and the call to `build` is at line 160